sale_purchase_force_vendor/models/stock_move.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _prepare_procurement_values(self):
+        res = super()._prepare_procurement_values()
+        if self.sale_line_id.vendor_id:
+            addr = self.sale_line_id.vendor_id.address_get(["invoice"])
+            supplier = self.env["res.partner"].browse(addr["invoice"])
+            product = self.product_id
+            res["supplierinfo_id"] = product.with_company(
```

Comment:
This seems duplicated with sale.order.line code

Reply:
You are right, I have removed the code and used the sale.order.line function.